### PR TITLE
CommonErrorsKata Assignment

### PR DESCRIPTION
**Pull request by wjarvis90**
Please review pull request, and let me know if any changes are needed.

### CODE REVIEW

File: CommonErrorsKata/QuizForm.cs [hunk 1]
```diff
@@ -10,21 +10,22 @@ namespace CommonErrorsKata
...
         {
             InitializeComponent();
-            synchronizationContext = SynchronizationContext.Current;
-            files = System.IO.Directory.GetFiles(Environment.CurrentDirectory +  @"..\..\..\ErrorPics");
-            possibleAnswers = new string[] { "Missing File", "null instance", "divide by zero" };
-            lstAnswers.DataSource = possibleAnswers;
-            answerQueue = new AnswerQueue<TrueFalseAnswer>(15);
+            _synchronizationContext = SynchronizationContext.Current;
+            _files = Directory.GetFiles(Environment.CurrentDirectory + @"..\..\..\ErrorPics");
+            _possibleAnswers = new[] { "Missing File", "null instance", "divide by zero" };
```
Comment (codytwinton) [inline CommonErrorsKata/QuizForm.cs]: Line 25 is no longer necessary because of line 26. Delete line 25.

File: CommonErrorsKata/QuizForm.cs [hunk 2]
```diff
@@ -10,21 +10,22 @@ namespace CommonErrorsKata
...
+            _possibleAnswers = _files.Select(f => Path.GetFileName(f)?.Replace(".png", " ")).ToArray();
+            lstAnswers.DataSource = _possibleAnswers;
+            _answerQueue = new AnswerQueue<TrueFalseAnswer>(15);
```
Comment (codytwinton) [inline CommonErrorsKata/QuizForm.cs]: Move 15 to a readonly variable. Don't want magic numbers.

File: CommonErrorsKata/QuizForm.cs [hunk 3]
```diff
@@ -44,36 +45,47 @@ public CommonErrorsForm()
...
+            if (selected != null && selected == _visibleImagePath)
+            {
+                _answerQueue.Enqueue(new TrueFalseAnswer(true));
+            }
+            else
+            {
+                _answerQueue.Enqueue(new TrueFalseAnswer(false));
+            }
+            var tokens = _visibleImagePath.Split(' ');
+            _answerQueue.Enqueue(new TrueFalseAnswer(true));
```
Comment (codytwinton) [inline CommonErrorsKata/QuizForm.cs]: Lines 58 and 59 are no longer necessary and break functionality. Delete them.

File: CommonErrorsKata/QuizForm.cs [hunk 4]
```diff
@@ -44,36 +45,47 @@ public CommonErrorsForm()
 
         private void LstAnswers_Click(object sender, EventArgs e)
         {
-            i = 100;
-            var tokens = currentBaseName.Split(' ');
-            //TODO:  Figure out what is a valid answer.
-            answerQueue.Enqueue(new TrueFalseAnswer(true));
+            _time = 100;
+            var selected = _possibleAnswers[lstAnswers.SelectedIndex];
+            if (selected != null && selected == _visibleImagePath)
```
Comment (codytwinton) [inline CommonErrorsKata/QuizForm.cs]: Try passing your conditional to your new TrueFalseAnswer instance.

File: CommonErrorsKata/QuizForm.cs [hunk 5]
```diff
@@ -44,36 +45,47 @@ public CommonErrorsForm()
...
                 Application.Exit();
                 return;
             }
-            label1.Text = answerQueue.Grade.ToString() + "%";
-            var file = files.GetRandom();
-            currentBaseName= Path.GetFileName(file);
+            label1.Text = _answerQueue.Grade.ToString() + "%";
+            var file = _files.GetRandom();
+            _visibleImagePath = Path.GetFileName(file)?.Replace(".png", " ");
+            _visibleImagePath = Path.GetFileName(file);
```
Comment (codytwinton) [inline CommonErrorsKata/QuizForm.cs]: Delete line 74.

File: CommonErrorsKata/QuizForm.cs [hunk 6]
```diff
@@ -44,36 +45,47 @@ public CommonErrorsForm()
...
+            }
+            var tokens = _visibleImagePath.Split(' ');
+            _answerQueue.Enqueue(new TrueFalseAnswer(true));
             Next();
         }
 
         private void Next()
         {
-            if (answerQueue.Count == 15 && answerQueue.Grade >= 98)
+            if (_answerQueue.Count == 4 && _answerQueue.Grade >= 98)
```
Comment (codytwinton) [inline CommonErrorsKata/QuizForm.cs]: Why 4?

**Comment by wjarvis90** [comment]
Pushed changes, please let me know if there are any changes needed.

**Pull request closed without merging**